contracts/modules/governance/NXMaster.sol

```diff
@@ -47,152 +45,118 @@ contract NXMaster is INXMMaster, Governed {
-    emergencyPaused.push(EmergencyPause(_pause, now, _by));
-    if (_pause == false) {
-      IClaims c1 = IClaims(allContractVersions["CL"]);
-      c1.submitClaimAfterEPOff(); // Process claims submitted while EP was on
-      c1.startAllPendingClaimsVoting(); // Resume voting on all pending claims
-    }
-  }
+    contractAddresses[contractCode] = address(uint160(newInternalContract));
+    contractsActive[newInternalContract] = true;
 
-  ///@dev update time in seconds for which emergency pause is applied.
-  function updatePauseTime(uint _time) public {
-    require(isInternal(msg.sender), "Not internal call.");
-    pauseTime = _time;
+    MasterAware up = MasterAware(contractAddresses[contractCode]);
+    up.changeMasterAddress(address(this));
+    up.changeDependentContractAddress();
   }
 
   /// @dev upgrades multiple contracts at a time
   function upgradeMultipleContracts(
-    bytes2[] memory _contractsName,
-    address payable[] memory _contractsAddress
+    bytes2[] memory _contractCodes,
+    address payable[] memory newAddresses
   )
   public
-  onlyAuthorizedToGovern
+  // onlyAuthorizedToGovern
   {
-    require(_contractsName.length == _contractsAddress.length, "Array length should be equal.");
+    require(_contractCodes.length == newAddresses.length, "NXMaster: Array length should be equal.");
 
-    for (uint i = 0; i < _contractsName.length; i++) {
+    for (uint i = 0; i < _contractCodes.length; i++) {
+      address payable newAddress = newAddresses[i];
+      bytes2 code = _contractCodes[i];
+      require(newAddress != address(0), "NXMaster: contract address is 0");
 
-      address payable newAddress = _contractsAddress[i];
-      require(newAddress != address(0), "NULL address is not allowed.");
-      require(isUpgradable[_contractsName[i]], "Contract should be upgradable.");
+      if (isProxy[code]) {
+        OwnedUpgradeabilityProxy proxy = OwnedUpgradeabilityProxy(contractAddresses[code]);
+        proxy.upgradeTo(newAddress);
+        continue;
+      }
 
-      if (_contractsName[i] == "QT") {
-        IQuotation qt = IQuotation(allContractVersions["QT"]);
-        qt.transferAssetsToNewContract(newAddress);
+      if (isUpgradable[code]) {
+        replaceContract(code, newAddress);
+        continue;
+      }
 
-      } else if (_contractsName[i] == "CR") {
-        ITokenController tc = ITokenController(getLatestAddress("TC"));
-        tc.addToWhitelist(newAddress);
-        tc.removeFromWhitelist(allContractVersions["CR"]);
-        IClaimsReward cr = IClaimsReward(allContractVersions["CR"]);
-        cr.upgrade(newAddress);
+      revert("NXMaster: non-existant contract code");
+    }
 
-      } else if (_contractsName[i] == "P1") {
-        IPool p1 = IPool(allContractVersions["P1"]);
-        p1.upgradeCapitalPool(newAddress);
-      }
+    for (uint i = 0; i < contractCodes.length; i++) {
+      contractsActive[contractAddresses[contractCodes[i]]] = true;
```

Comment:
This line is redundant. All contracts are already marked active. Proxies on creation and replaceable/upgradable in replaceContract.